staging/src/k8s.io/dynamic-resource-allocation/structured/allocator.go

```diff
@@ -680,7 +687,13 @@ func (alloc *allocator) allocateOne(r deviceIndices, allocateSubRequest bool) (b
 	claim := alloc.claimsToAllocate[r.claimIndex]
 	if r.requestIndex >= len(claim.Spec.Devices.Requests) {
 		// Done with the claim, continue with the next one.
-		return alloc.allocateOne(deviceIndices{claimIndex: r.claimIndex + 1}, false)
+		success, err := alloc.allocateOne(deviceIndices{claimIndex: r.claimIndex + 1}, false)
+		if errors.Is(err, errAllocationResultMaxSizeExceeded) {
+			// We don't need to propagate this further, since the allocation size
+			// limit is per claim.
```

Comment:
So we failed to allocate the current claim because the result would be too large. Under which circumstances does it make sense to undo some prior work and to try allocating this claim again?

Isn't it *always* going to fail because of the size limit? Thinking aloud here... no, it isn't.

If the allocation of the first claim influences which sub-request in the second claim gets chosen, then the number of devices in the second claim can be different during a retry.

Worth calling out here:

```suggestion
			// We don't need to propagate this further because
			// this is not a fatal error. Retrying the claim under
			// different circumstances may succeed if it uses
			// subrequests and changing the allocation of some
			// prior claim enables allocating a subrequest here
			// which needs less devices.
```

Reply:
Agree. I've updated the comment.